ext/session/session.c

```diff
@@ -2379,7 +2382,7 @@ PHP_FUNCTION(session_create_id)
 	}
 
 	if (prefix && ZSTR_LEN(prefix)) {
-		if (php_session_valid_key(ZSTR_VAL(prefix)) == FAILURE) {
+		if (php_session_valid_key(prefix) == FAILURE) {
```

Comment:
You could move the nul-byte check into the parameter parsing using the `P` specifier.

Reply:
I'm sorry but I don't know what is `P` specifier so I chose your other idea.

Reply:
See: https://github.com/php/php-src/blob/master/docs-old/parameter-parsing-api.md#type-specifiers

Reply:
Thanks for the explanation. Which way should I choose then? I'm open to the choice.

Reply:
I would move it into the ZPP check, as then it throws a ValueError instead of a warning.

Reply:
Thanks. It's a much cleaner solution and simpler. Applied changes